src/main/java/org/opensrp/util/IdGeneratorUtil.java

```diff
@@ -0,0 +1,48 @@
+package org.opensrp.util;
+
+public class IdGeneratorUtil {
+
+	/**
+	 * Converts a String back to an long based on the passed base characters
+	 * @should convert from string in base character set to long
+	 */
+	public static long convertFromBase(String s, char[] baseCharacters) {
+		long ret = 0;
+		char[] inputChars = s.toCharArray();
+		long multiplier = 1;
+		for (int i = inputChars.length-1; i>=0; i--) {
+			int index = -1;
+			for (int j=0; j<baseCharacters.length; j++) {
+				if (baseCharacters[j] == inputChars[i]) {
+					index = j;
+				}
+			}
+			if (index == -1) {
+				throw new RuntimeException("Invalid character " + inputChars[i] + " found in " + s);
```

Comment:
Use IllegalArgumentException instead of RuntimeException for invalid values being passed. Use this for all such instances in the PR